Bottom: e277b5d4a71170de250c6c135d9810d44375cea7
Top:    143c699a373dc8318cadc73f45fe932ee2c97b48
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-06 17:09:48 +0100

Refresh of workaround-crash-freeing

---

diff --git a/src/polygon1.c b/src/polygon1.c
index cb7f62a..9df24ee 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -3633,11 +3633,25 @@ poly_DelContour (PLINE ** c)
   for (cur = PREV_EDGE (&(*c)->head); cur != &(*c)->head; cur = prev)
     {
       prev = PREV_EDGE (cur);
-      if (cur->cvc_next != NULL)
-	{
-	  free (cur->cvc_next);
-	  free (cur->cvc_prev);
-	}
+
+      if (cur->cvc_next == -1)
+        {
+          fprintf (stderr, "WHY DID THIS HAPPEN?? cvc_next == -1 in poly_DelContour() !\n");
+        }
+      else if (cur->cvc_next != NULL)
+        {
+          free (cur->cvc_next);
+        }
+
+      if (cur->cvc_prev == -1)
+        {
+          fprintf (stderr, "WHY DID THIS HAPPEN?? cvc_prev == -1 in poly_DelContour() !\n");
+        }
+      else if (cur->cvc_prev != NULL)
+        {
+          free (cur->cvc_prev);
+        }
+
       g_slice_free (VNODE, cur);
     }
   if ((*c)->head.cvc_next != NULL)
